lib/addons/jupyterhub/index.ts

```diff
@@ -0,0 +1,52 @@
+import { Construct } from "constructs";
+import { ClusterInfo } from '../../spi';
+import { createNamespace } from '../../utils';
+import { HelmAddOn, HelmAddOnProps, HelmAddOnUserProps } from '../helm-addon';
+
+/**
+ * Configuration options for the add-on.
+ */
+ type JupyterHubAddOnProps = HelmAddOnUserProps;
+
+const JUPYTERHUB = 'jupyterhub';
+const RELEASE = 'blueprints-addon-jupyterhub';
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps: HelmAddOnProps = {
+    name: JUPYTERHUB,
+    namespace: JUPYTERHUB,
+    version: '1.2.0',
+    chart: JUPYTERHUB,
+    release: RELEASE,
+    repository: 'https://jupyterhub.github.io/helm-chart/',
+};
+
+/**
+ * Implementation of the JupyterHub add-on
+ */
+export class JupyterHubAddOn extends HelmAddOn {
+
+    readonly options: JupyterHubAddOnProps;
+
+    constructor(props?: JupyterHubAddOnProps) {
+        super({...defaultProps, ...props});
+        this.options = this.props;
+    }
+
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let values = this.options.values ?? {};
+
+        // Create Namespace
+        const ns = createNamespace(JUPYTERHUB, cluster, true, true);
```

Comment:
Is deploying to an existing namespace not an option?

Reply:
Not addressed? Why namespace is hardcoded?

Reply:
addressed.

Reply:
addressed.